S3UTILS-193 bump bucketclient

jonathan-gramain opened the pull request:
Also added a missing direct dependency on "mongodb" (see commit message for details)

jonathan-gramain committed:
bump bucketclient

Issue: S3UTILS-193

jonathan-gramain committed:
Bump project version

Issue: S3UTILS-193

bert-e commented:
# Hello jonathan-gramain,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/s3utils/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Request integration branches



Waiting for integration branch creation to be requested by the user.

To request integration branches, please comment on this pull request with the following command:

```
/create_integration_branches
```

Alternatively, the `/approve` and `/create_pull_requests` commands will automatically
create the integration branches.

jonathan-gramain commented:
/create_integration_branches

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/1.15/bugfix/S3UTILS-193-bump-bucketclient` with contents from `bugfix/S3UTILS-193-bump-bucketclient`
and `development/1.15`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/1.15/bugfix/S3UTILS-193-bump-bucketclient origin/development/1.15
 $ git merge origin/bugfix/S3UTILS-193-bump-bucketclient
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/1.15/bugfix/S3UTILS-193-bump-bucketclient
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/1.16/bugfix/S3UTILS-193-bump-bucketclient` with contents from `w/1.15/bugfix/S3UTILS-193-bump-bucketclient`
and `development/1.16`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/1.16/bugfix/S3UTILS-193-bump-bucketclient origin/development/1.16
 $ git merge origin/w/1.15/bugfix/S3UTILS-193-bump-bucketclient
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/1.16/bugfix/S3UTILS-193-bump-bucketclient
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #8e2b3338642b3a1772940e14eade006dc5cf39e8 on the integration branch
`w/1.15/bugfix/S3UTILS-193-bump-bucketclient` is merging a branch which is neither the current
branch `bugfix/S3UTILS-193-bump-bucketclient` nor the development branch
`development/1.15`.

It is likely due to a rebase of the branch `bugfix/S3UTILS-193-bump-bucketclient` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue S3UTILS-193 contains:


* `1.14.21`

* `1.15.12`

* `1.16.1`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.15.12`

* `1.16.1`


Please check the `Fix Version/s` of S3UTILS-193, or the target
branch of this pull request.




*The following options are set:* **create_integration_branches**

bert-e commented:
# History mismatch



Merge commit #8e2b3338642b3a1772940e14eade006dc5cf39e8 on the integration branch
`w/1.16/bugfix/S3UTILS-193-bump-bucketclient` is merging a branch which is neither the current
branch `bugfix/S3UTILS-193-bump-bucketclient` nor the development branch
`development/1.16`.

It is likely due to a rebase of the branch `bugfix/S3UTILS-193-bump-bucketclient` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **create_integration_branches**

jonathan-gramain committed:
Add a missing direct dependency to mongodb

The "mongodb" module is imported directly by some tests, so it has to
be declared in package.json with a compatible version. I used the same
version than in the current direct dependency of arsenal.

Not having this direct dependency caused test failures due to an older
mongodb 3 version being hoisted instead for arcane reasons (from
another indirect dependency). This caused BSON version incompatibility
errors.

issue: S3UTILS-193

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue S3UTILS-193 contains:


* `1.15.12`

* `1.16.1`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `1.14.21`

* `1.15.12`

* `1.16.1`


Please check the `Fix Version/s` of S3UTILS-193, or the target
branch of this pull request.




*The following options are set:* **create_integration_branches**

jonathan-gramain commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/1.15/bugfix/S3UTILS-193-bump-bucketclient` with contents from `bugfix/S3UTILS-193-bump-bucketclient`
and `development/1.15`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/1.15/bugfix/S3UTILS-193-bump-bucketclient origin/development/1.15
 $ git merge origin/bugfix/S3UTILS-193-bump-bucketclient
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/1.15/bugfix/S3UTILS-193-bump-bucketclient
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/1.16/bugfix/S3UTILS-193-bump-bucketclient` with contents from `w/1.15/bugfix/S3UTILS-193-bump-bucketclient`
and `development/1.16`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/1.16/bugfix/S3UTILS-193-bump-bucketclient origin/development/1.16
 $ git merge origin/w/1.15/bugfix/S3UTILS-193-bump-bucketclient
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/1.16/bugfix/S3UTILS-193-bump-bucketclient
```





*The following options are set:* **create_integration_branches**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers











*The following options are set:* **create_integration_branches**

jonathan-gramain commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/1.14`

* :heavy_check_mark: `development/1.15`

* :heavy_check_mark: `development/1.16`



The following branches have **NOT** changed:

* `development/1.13`
* `development/1.4`



Please check the status of the associated issue S3UTILS-193.

Goodbye jonathan-gramain.




*The following options are set:* **approve, create_integration_branches**
